nms/app/packages/magmalte/app/views/equipment/GatewayYMLConfig.js

```diff
@@ -0,0 +1,170 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow
+ * @format
+ */
+import type {WithAlert} from '@fbcnms/ui/components/Alert/withAlert';
+
+import withAlert from '@fbcnms/ui/components/Alert/withAlert';
+
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import ListItemText from '@material-ui/core/ListItemText';
+import LoadingFiller from '@fbcnms/ui/components/LoadingFiller';
+import MagmaV1API from '@fbcnms/magma-api/client/WebClient';
+import MenuItem from '@material-ui/core/MenuItem';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
+import React from 'react';
+import Select from '@material-ui/core/Select';
+import Text from '../../theme/design-system/Text';
+import TextareaAutosize from 'react-textarea-autosize';
+import nullthrows from '@fbcnms/util/nullthrows';
+import useMagmaAPI from '@fbcnms/ui/magma/useMagmaAPI';
+
+import {makeStyles} from '@material-ui/styles';
+import {useCallback, useState} from 'react';
+import {useRouter, useSnackbar} from '@fbcnms/ui/hooks';
+
+const useStyles = makeStyles(theme => ({
+  dashboardRoot: {
+    margin: theme.spacing(3),
+    flexGrow: 1,
+  },
+  ymlEditor: {
+    minWidth: '80%',
+    padding: '10px',
+  },
+  description: {
+    marginRight: '15px',
+  },
+}));
+
+// td-agent-bit - template file
+type Props = {...WithAlert};
+function GatewayConfigYml(props: Props) {
+  const RUNNING_SERVICES = [
+    'policydb',
+    'control_proxy',
+    'mobilityd',
+    'smsd',
+    'pipelined',
+    'sessiond',
+    'redis',
+    'dnsd',
+    'mme',
+    'directoryd',
+    'eventd',
+    'enodebd',
+    'state',
+    'subscriberdb',
+    'magmad',
+    'health',
+    'ctraced',
+  ];
+  const {match} = useRouter();
+  const networkId = nullthrows(match.params.networkId);
+  const classes = useStyles();
+  const gatewayId: string = nullthrows(match.params.gatewayId);
+  const [selectedService, setSelectedService] = useState(RUNNING_SERVICES[0]);
+  const SERVICE_CONTENT = `cat /etc/magma/${selectedService}.yml`;
+  const serviceParams = {
+    command: 'bash',
+    params: {
+      shell_params: [`-c '${SERVICE_CONTENT}'`],
+    },
+  };
+  const [serviceConfig, setServiceConfig] = useState<string>('');
+
+  const onSave = () => {
+    const config = serviceConfig.replace(/"/g, '\\"');
+    props
+      .confirm('Are you sure you want to save this config?')
+      .then(async confirm => {
+        if (!confirm) return;
+        await MagmaV1API.postNetworksByNetworkIdGatewaysByGatewayIdCommandGeneric(
+          {
+            networkId,
+            gatewayId,
+            parameters: {
+              command: 'bash',
+              params: {
+                // $FlowIssue[incompatible-call]
+                shell_params: [
+                  `-c 'printf -- "${config}" > /etc/magma/test.yml'`,
```

Comment:
Avoid single quotes `'` like the plague.

https://www.gnu.org/software/bash/manual/html_node/Single-Quotes.html

```suggestion
                  `"cat >/etc/magma/test.yml <<EOL ${config} EOL "`,
```

With the suggested change, you're never enclosing strings in single quotes.